src/libsyntax/parse/lexer/tokentrees.rs

```diff
@@ -121,13 +124,20 @@ impl<'a> TokenTreesReader<'a> {
                     // Correct delimiter.
                     token::CloseDelim(d) if d == delim => {
                         let (open_brace, open_brace_span) = self.open_braces.pop().unwrap();
+                        let close_brace_span = self.token.span;
+
+                        if close_brace_span.lo() == open_brace_span.hi() {
```

Comment:
It seems to me that it would make more sense to check whether `tts` is empty instead of comparing spans. That way we also catch cases like `{ }` and `{\n}`.

Reply:
Thanks for the advice and I'll to change it.